### ISSUE TITLE
Support C++ style alignment for structs

### ISSUE THREAD

**Issue by JeffRous** [ispc/ispc#2145]
Currently there's not a good way to specify struct alignment in ISPC. This can lead to bugs and suboptimal code generation when needing to work with data passed in from C++. Would like something similar to MSVC `__declspec(align(n))` and GCC/Clang `__attribute__((aligned(n)))`.

`assume()` works but isn't the best solution since it requires a statement for every pointer. Soa<> notation also works but only for struct members and is non-obvious.

**Comment by pbrubaker** [comment]
We might want to consider using `alignas()` from C++ 11.

https://en.cppreference.com/w/cpp/language/alignas

**Comment by aneshlya** [comment]
After adding attributes to the language this is becoming possible. @nurmukhametov, what do you think?

**Comment by nurmukhametov** [comment]
> After adding attributes to the language this is becoming possible. [@nurmukhametov](https://github.com/nurmukhametov), what do you think?

Yes, I think so. @JeffRous, if you have any current examples of code where you have worked around this problem, it would be helpful.

**Comment by nurmukhametov** [comment]
#3302 enabled the aligned attribute for variables and struct types as a whole.
